### DISCUSSION TITLE
Using Viewport Height and Width as Image Height and Width Doesn't work

### DISCUSSION THREAD

**Discussion by ebitogu** [vercel/next.js#22294] [Help]
**What version of Next.js are you using?**

10.0.7

**What version of Node.js are you using?**

6.14.8

**What browser are you using?**

Chrome

**What operating system are you using?**

Windows

**How are you deploying your application?**

Vercel

**Describe the Bug**

I wish to constrain an Image using the View Port's Width and Height as below as against using Pixel.

```<Image src="/images/trap_two.jpg" width="100vw" height="80vh" />```

**Expected Behavior**

I expect that the Image would take up the full width of the page and 80% of the View Port Height.

But that is not the case. It never stretches at all.

**To Reproduce**

```<Image src="/images/trap_two.jpg" width="100vw" height="80vh" />```

**Comment by michrome** [comment]
`width` and `height` should contain width and height of **the `src` image file**, as integers and without a unit.

For example, if `/images/trap_two.jpg` is 1024px x 768px then you should use `<Image src="/images/trap_two.jpg" width={1024} height={768} />`.

Use a container in combination with [`layout`](https://nextjs.org/docs/api-reference/next/image#layout) to constrain the image e.g.

```js
  return (
    <div style={{ width: "100vw", height: "80vh" }}>
      <Image
        src="/images/trap_two.jpg"
        alt="Image description here"
        width={1024}
        height={768}
      />
    </div>
  );
```

**Comment by nico-bachner** [comment]
I think it's also worth noting that stretching can be achieved using `layout="fill"` instead of the `width` and `height` attributes. 

In the case of @michrome's example, you would get the following:

``` jsx
return (
  <div style={{ width: "100vw", height: "80vh" }}>
    <Image
      src="/images/trap_two.jpg"
      alt="Image description here"
      layout="fill"
    />
  </div>
);
```

According to the docs, this should make your image stretch in order to fill the available space

However, last I checked, `layout="fill"` did not fit the image inside the container element, but rather the entire viewport.
